### ISSUE TITLE
Orphan code fragment

### ISSUE THREAD

**Issue by fredrikr** [DavidKinder/Inform6#28]
In some z3 story files, I get an orphan code fragment at the end of Zcode memory. This is what it looks like in "TXD -d -n" output:

orphan code fragment:

 5006:  4e 97 e5                INSERT_OBJ      G87,"lgwitsm"
 5009:  a5 62                   INC             [G52]
 500b:  9a e5                   PRINT_OBJ       "lgwitsm"
 500d:  a5 28                   INC             [G18]
 500f:  d8 e4 a5                MOD             -> G95
 5012:  71 58 e4 a5             GET_PROP        G48,Gd4 -> G95
 5016:  ea a5 26 9c cc a5       SPLIT_WINDOW    G16,G8c,#cc,#a5
 501c:  ba                      QUIT            

The code for this game contains a single SPLIT_WINDOW statement, and this is not it. This must be something which the compiler has thrown in.

I'm using the Inform 6.34 Win32 binary available at if-archive, with PunyInform.

**Comment by erkyrath** [comment]
I tried compiling library_of_horror.inf, minimal.inf, and cloak.inf from the PunyInform repo and didn't see this. Can you provide a test case?

**Comment by fredrikr** [comment]
Compiling nada.inf to v3, with or without -e, gives this orphan code.

**Comment by erkyrath** [comment]
Got it, thanks.

I first thought that this had to do with the OMIT_UNUSED_ROUTINES switch, so I turned that off. But I still see the "orphaned code" output from txd.

Then I realized that the fragment address (without OMIT_UNUSED_ROUTINES) is 52ae. If you compile with the -z switch, you'll see that 52ae is the start of the strings segment. So I think this is a txd bug, not a compile bug! txd is slipping into string space while decompiling routines.

I took a quick look at the txd code to see if this was an obvious bug, but it's pretty opaque.
